### DISCUSSION TITLE
Adding DiscordSocketClient to a service?

### DISCUSSION THREAD

**Discussion by JohnSpade117** [Hawxy/Discord.Addons.Hosting#15] [Q&A] [answered]
Good evening,
Is there a way to use the DiscordSocketClient to add a service? Attached is an image of what I'm talking about. The service requires the discordsocketclient.

Thank you so much for your help!

![image1](https://user-images.githubusercontent.com/83733163/117215612-50b93b80-adb3-11eb-8b0c-17d95f1b49bc.PNG)

**Comment by Hawxy** [comment, marked as answer]
Hello,

Use the factory lambda overload on the `AddSingleton` call like so:
 ` .AddSingleton(x => new InteractivityService(x.GetRequiredService<DiscordSocketClient>(), TimeSpan.FromMinutes(3)))`
